packages/edit-site/src/hooks/use-theme-style-variations/use-theme-style-variations-by-property.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { mergeBaseAndUserConfigs } from '../../components/global-styles/global-styles-provider';
+import cloneDeep from '../../utils/clone-deep';
+
+/**
+ * Returns a new object with only the properties specified in `properties`.
+ *
+ * @param {Object} object   The object to filter
+ * @param {Object} property The property to filter by
+ * @return {Object} The merged object.
+ */
+export const filterObjectByProperty = ( object, property ) => {
+	if ( ! object ) {
+		return {};
```

Comment:
Should this be returning a new object, or return `object` as is? Apologies if this is already covered!

Reply:
Probably object so the recursion works. 

I'm adding unit tests for this function so we can check these scenarios.

Thank you for checking it out! 🙇🏻

Reply:
> Probably object so the recursion works.
> I'm adding unit tests for this function so we can check these scenarios.

Thanks! The main thing to check is that we don't accidentally coerce any falsy leaves in a tree to object values 🙂